scene/2d/navigation_obstacle_2d.cpp

```diff
@@ -0,0 +1,196 @@
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "navigation_obstacle_2d.h"
+
+#include "scene/2d/collision_shape_2d.h"
+#include "scene/2d/navigation_2d.h"
+#include "scene/2d/physics_body_2d.h"
+#include "servers/navigation_2d_server.h"
+
+void NavigationObstacle2D::_bind_methods() {
+	ClassDB::bind_method(D_METHOD("set_navigation", "navigation"), &NavigationObstacle2D::set_navigation_node);
+	ClassDB::bind_method(D_METHOD("get_navigation"), &NavigationObstacle2D::get_navigation_node);
+	ClassDB::bind_method(D_METHOD("set_estimate_radius", "estimate_radius"), &NavigationObstacle2D::set_estimate_radius);
+	ClassDB::bind_method(D_METHOD("is_radius_estimated"), &NavigationObstacle2D::is_radius_estimated);
+	ClassDB::bind_method(D_METHOD("set_radius", "radius"), &NavigationObstacle2D::set_radius);
+	ClassDB::bind_method(D_METHOD("get_radius"), &NavigationObstacle2D::get_radius);
+
+	ADD_PROPERTY(PropertyInfo(Variant::BOOL, "estimate_radius"), "set_estimate_radius", "is_radius_estimated");
+	ADD_PROPERTY(PropertyInfo(Variant::REAL, "radius", PROPERTY_HINT_RANGE, "0.01,500,0.01"), "set_radius", "get_radius");
+}
+
+void NavigationObstacle2D::_validate_property(PropertyInfo &p_property) const {
+	if (p_property.name == "radius") {
+		if (estimate_radius) {
+			p_property.usage = PROPERTY_USAGE_NOEDITOR;
+		}
+	}
+}
+
+void NavigationObstacle2D::_notification(int p_what) {
+	switch (p_what) {
+		case NOTIFICATION_READY: {
+			initialize_agent();
+			parent_node2d = Object::cast_to<Node2D>(get_parent());
+
+			// Search the navigation node and set it
+			{
+				Navigation2D *nav = nullptr;
+				Node *p = get_parent();
+				while (p != nullptr) {
+					nav = Object::cast_to<Navigation2D>(p);
+					if (nav != nullptr) {
+						p = nullptr;
+					} else {
+						p = p->get_parent();
+					}
+				}
+
+				set_navigation(nav);
+			}
+
+			set_physics_process_internal(true);
+		} break;
+		case NOTIFICATION_EXIT_TREE: {
+			set_navigation(nullptr);
+			set_physics_process_internal(false);
+			request_ready(); // required to solve an issue with losing the navigation
+		} break;
+		case NOTIFICATION_PARENTED: {
+			parent_node2d = Object::cast_to<Node2D>(get_parent());
+			reevaluate_agent_radius();
+		}
+		case NOTIFICATION_UNPARENTED: {
+			parent_node2d = nullptr;
+		}
```

Comment:
These two `case`s are missing a `break` 😯

Reply:
How come the compiler didn't warn about this? It's fallthrough but there's no fallthrough statement.